plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/GlueTableOperations.java

```diff
@@ -0,0 +1,236 @@
+import io.trino.spi.TrinoException;
+import io.trino.spi.connector.ConnectorSession;
+import io.trino.spi.connector.TableNotFoundException;
+import org.apache.hadoop.hive.metastore.TableType;
+import org.apache.iceberg.TableMetadata;
+import org.apache.iceberg.exceptions.CommitFailedException;
+import org.apache.iceberg.io.FileIO;
+
+import java.util.Map;
+import java.util.Optional;
+
+import static com.google.common.base.Preconditions.checkState;
+import static io.trino.plugin.hive.ViewReaderUtil.PRESTO_VIEW_FLAG;
+import static io.trino.plugin.iceberg.IcebergErrorCode.ICEBERG_COMMIT_ERROR;
+import static io.trino.plugin.iceberg.IcebergErrorCode.ICEBERG_INVALID_METADATA;
+import static java.lang.String.format;
+import static java.util.Locale.ENGLISH;
+import static java.util.Objects.requireNonNull;
+import static org.apache.hadoop.hive.metastore.TableType.EXTERNAL_TABLE;
+import static org.apache.iceberg.BaseMetastoreTableOperations.ICEBERG_TABLE_TYPE_VALUE;
+import static org.apache.iceberg.BaseMetastoreTableOperations.TABLE_TYPE_PROP;
+
+public class GlueTableOperations
+        extends AbstractIcebergTableOperations
+{
+    private static final Logger log = Logger.get(GlueTableOperations.class);
+
+    private final AWSGlueAsync glueClient;
+    private final GlueMetastoreStats stats;
+    private final String catalogId;
+
+    protected GlueTableOperations(
+            AWSGlueAsync glueClient,
+            GlueMetastoreStats stats,
+            String catalogId,
+            FileIO fileIo,
+            ConnectorSession session,
+            String database,
+            String table,
+            Optional<String> owner,
+            Optional<String> location)
+    {
+        super(fileIo, session, database, table, owner, location);
+        this.glueClient = requireNonNull(glueClient, "glueClient is null");
+        this.stats = requireNonNull(stats, "stats is null");
+        this.catalogId = catalogId;
+    }
+
+    @Override
+    protected String getRefreshedLocation()
+    {
+        return stats.getGetTable().call(() -> {
+            Table table = getTable();
+
+            if (isPrestoView(table) && isHiveOrPrestoView(table)) {
+                // this is a Presto Hive view, hence not a table
+                throw new TableNotFoundException(getSchemaTableName());
+            }
+            if (!isIcebergTable(table)) {
+                throw new UnknownTableTypeException(getSchemaTableName());
+            }
+
+            String metadataLocation = table.getParameters().get(METADATA_LOCATION);
+            if (metadataLocation == null) {
+                throw new TrinoException(ICEBERG_INVALID_METADATA, format("Table is missing [%s] property: %s", METADATA_LOCATION, getSchemaTableName()));
+            }
+            return metadataLocation;
+        });
+    }
+
+    @Override
+    protected void commitNewTable(TableMetadata metadata)
+    {
+        String newMetadataLocation = writeNewMetadata(metadata, version + 1);
```

Comment:
why `+1` for a new table?

Reply:
I'm going to switch this to verify `version =-= -1` and pass `0` to `writeNewMetadata`